### PR TITLE
Com.GitHub.ytdl_gui.youtubedl_gui

### PR DESCRIPTION
**Pull request by advaithm**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by advaithm** [comment]
bot, build com.github.ytdl_gui.youtubedl_gui

**Comment by advaithm** [comment]
bot, build com.github.ytdl_gui.youtubedl_gui

File: com.github.ytdl_gui.youtubedl_gui.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: youtubedl-gui
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=host
```
Comment (nedrichards) [inline com.github.ytdl_gui.youtubedl_gui.yaml:11]: this permission is too broad without further justification. Can the app just run using the portal based access and not need any permission at all? Or be scoped to xdg-download by default?

Comment (advaithm) [reply com.github.ytdl_gui.youtubedl_gui.yaml:11]: the app opens to xdg-download by default, it should be possible to reduce the permission to just that

**Comment by advaithm** [comment]
bot, build com.github.ytdl_gui.youtubedl_gui

**Comment by advaithm** [comment]
bot, build com.github.ytdl_gui.youtubedl_gui

File: com.github.ytdl_gui.youtubedl_gui.yaml (hunk 1 above)
Comment (advaithm) [reply com.github.ytdl_gui.youtubedl_gui.yaml]: I spoke to the developer we can use portal based access

File: com.github.ytdl_gui.youtubedl_gui.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.github.ytdl_gui.youtubedl_gui</id>
```
Comment (barthalion) [inline com.github.ytdl_gui.youtubedl_gui.metainfo.xml:3]: This ID seems wrong to me? It should be `io.github.JaGoLi.ytdl_gyi`.

File: com.github.ytdl_gui.youtubedl_gui.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+app-id: com.github.ytdl_gui.youtubedl_gui
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: youtubedl-gui
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
```
Comment (barthalion) [inline com.github.ytdl_gui.youtubedl_gui.yaml:9]:
```suggestion
  - --socket=fallback-x11
```

File: com.github.ytdl_gui.youtubedl_gui.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      - mv /app/share/icons/hicolor/256x256/apps/youtubedl-gui.png /app/share/icons/hicolor/256x256/apps/com.github.ytdl_gui.youtubedl_gui.png
+      - mv /app/share/icons/hicolor/32x32/apps/youtubedl-gui.png /app/share/icons/hicolor/32x32/apps/com.github.ytdl_gui.youtubedl_gui.png
+      - mv /app/share/icons/hicolor/512x512/apps/youtubedl-gui.png /app/share/icons/hicolor/512x512/apps/com.github.ytdl_gui.youtubedl_gui.png
+      - mv /app/share/icons/hicolor/16x16/apps/youtubedl-gui.png /app/share/icons/hicolor/16x16/apps/com.github.ytdl_gui.youtubedl_gui.png
+      - mv /app/share/applications/youtubedl-gui.desktop /app/share/applications/com.github.ytdl_gui.youtubedl_gui.desktop
+      - sed -i '3d' /app/share/applications/com.github.ytdl_gui.youtubedl_gui.desktop
+      - sed -i 's/Icon=youtubedl-gui/Icon=com.github.ytdl_gui.youtubedl_gui/g' /app/share/applications/com.github.ytdl_gui.youtubedl_gui.desktop
+      - sed -i 's/Exec=\/usr\/bin\/youtubedl-gui/Exec=youtubedl-gui/g' /app/share/applications/com.github.ytdl_gui.youtubedl_gui.desktop
```
Comment (barthalion) [inline com.github.ytdl_gui.youtubedl_gui.yaml:66]: You can use destkop-file-edit to do these changes.

File: com.github.ytdl_gui.youtubedl_gui.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+          url: https://ffmpeg.org/releases/ffmpeg-4.4.tar.xz
+          sha256: 06b10a183ce5371f915c6bb15b7b1fffbe046e8275099c96affc29e17645d909
+
+  - python3-youtube-dl.json
+
+  - name: youtubedl-gui
+    buildsystem: cmake
+    sources:
+        - type: archive
+          url: https://github.com/JaGoLi/ytdl-gui/archive/refs/tags/2.5.tar.gz
```
Comment (barthalion) [inline com.github.ytdl_gui.youtubedl_gui.yaml:52]: Consider adding section for https://github.com/flathub/flatpak-external-data-checker/ here

Comment (advaithm) [reply com.github.ytdl_gui.youtubedl_gui.yaml]: I don't quite understand how to do this

**Comment by advaithm** [comment]
bot, build io.github.JaGoLi.ytdl_gui

**Comment by barthalion** [comment]
bot, build io.github.JaGoLi.ytdl_gui

File: io.github.JaGoLi.ytdl_gui.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+app-id: io.github.JaGoLi.ytdl_gui
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: youtubedl-gui
+
+finish-args:
```
Comment (barthalion) [inline io.github.JaGoLi.ytdl_gui.yaml:7]: I think the way portals are used here is wonky so adding `--filesystem=home` may be the best choice.

Comment (advaithm) [reply io.github.JaGoLi.ytdl_gui.yaml:7]: okay, I'll add the permission

**Comment by advaithm** [comment]
bot, build io.github.JaGoLi.ytdl_gui

**Pull request closed without merging**